jobsdb/jobsdb.go

```diff
@@ -2421,7 +2419,7 @@ func (jd *HandleT) markClearEmptyResult(ds dataSetT, workspace string, stateFilt
 	}
 
 	// Safe Check , All parameter filters must be provided explicitly
-	if len(parameterFilters) != len(CacheKeyParameterFilters) {
+	if len(parameterFilters) != len(CacheKeyParameterFilters) && len(parameterFilters) != 0 {
```

Comment:
is length alone enough? are we checking that keys match elswhere?

Reply:
We check it in the below loop. If the length matches and the keys match then we go ahead